lib/blob/blobstore.c

```diff
@@ -3239,13 +3275,17 @@ blob_request_submit_rw_iov(struct spdk_blob *blob, struct spdk_io_channel *_chan
 
 			return;
 		}
-
-		is_allocated = blob_calculate_lba_and_lba_count(blob, offset, length, &lba, &lba_count);
+		if (read & ((ext_io_flags & FLAG_READ_ALLOCATED) == FLAG_READ_ALLOCATED)) {
```

Comment:
If we only set read flag on reads, might be faster to not check anything else?
```suggestion
		if (ext_io_flags & FLAG_READ_ALLOCATED) {
```

Reply:
Doesn't the flag already exist? SPDK_NVME_IO_FLAGS_UNWRITTEN_READ_FAIL

Reply:
Yes. I miss judge one point. ext_io_flag is set to `SPDK_NVME_IO_FLAGS_UNWRITTEN_READ_FAIL` in io-engine rather 1. We can use the same condition here. Changed the code. PTAL.